core/frontend/src/tools/ElementSetTool.ts

```diff
@@ -6,141 +6,51 @@
-  None,
-}
-
-/** Helps determine the action ModifyAgenda will take on the agenda elements after calling doFenceClip.
- * @internal
- */
-export enum ClipResult {
-  /** Tool does not support fence clip */
-  NotSupported,
-  /** Valid entries in ElementAgenda are new elements, no elemRef but correct modelRef. */
-  NewElements,
-  /** Valid entries in ElementAgenda reference original elemRef and modelRef. */
-  OriginalElements,
-}
-
-/** ElemSource specific failures.
- * @internal
- */
-export enum ErrorNums {
-  /** No fence is currently active */
-  NoFence,
-  /** No acceptable element(s) inside fence */
-  NoFenceElems,
-  /** No acceptable element(s) outside fence */
-  NoFenceElemsOutside,
-  /** No acceptable element(s) in selection set */
-  NoSSElems,
-  /** Located element rejected by tool filters */
-  NotSupportedElmType,
-}
-
-/** @internal */
-export enum HilitedState {
-  /**  this agenda is in an indeterminate state wrt hilite */
-  Unknown = 0,
-  /**  all of the entries in this agenda were hilited by a call to ElementAgenda.hilite */
-  Yes = 1,
-  /**  all of the entries in this agenda were un-hilited by a call to ElementAgenda.clearHilite */
-  No = 2,
-}
-
-/** @internal */
+import { NotifyMessageDetails, OutputMessagePriority } from "../NotificationManager";
+import { Pixel } from "../render/Pixel";
+import { SelectionSet } from "../SelectionSet";
+import { DecorateContext } from "../ViewContext";
+import { Viewport } from "../Viewport";
+import { ViewRect } from "../ViewRect";
+import { PrimitiveTool } from "./PrimitiveTool";
+import { SelectionMethod } from "./SelectTool";
+import { BeButton, BeButtonEvent, BeModifierKeys, CoreTools, EventHandled } from "./Tool";
+import { ToolAssistance, ToolAssistanceImage, ToolAssistanceInputMethod, ToolAssistanceInstruction, ToolAssistanceSection } from "./ToolAssistance";
+
+/** @alpha */
 export enum ModifyElementSource {
   /** The source for the element is unknown - not caused by a modification command. */
   Unknown = 0,
   /** The element is selected by the user. */
   Selected = 1,
   /** The element is processed because it is in the selection set. */
-  SelectionSet = 2,
-  /** The element is processed because it is passes the fence criteria. */
-  Fence = 3,
-  /** The element is processed because it belongs to the group of the selected element (for _FilterAgendaEntries only) */
-  Group = 4,
-  /** The element is selected by the user by drag selection or multi-selection using ctrl. */
-  DragSelect = 5,
+  SelectionSet = 2, // eslint-disable-line @typescript-eslint/no-shadow
+  /** The element is selected by the user using drag box or crossing line selection. */
+  DragSelect = 3,
 }
 
-/** @internal */
+/** @alpha */
 export interface GroupMark {
   start: number;
   source: ModifyElementSource;
 }
 
-/** @internal */
+/** @alpha */
 export class ElementAgenda {
-  public readonly elements: string[] = [];
+  public readonly elements: Id64Array = [];
```

Comment:
Member functions like `add` and `remove` do additional stuff with e.g. hilite and group marks. Is it cool if people can just modify this array directly?

Reply:
I'll add a comment that it's better not to change directly. The "get candidate" methods on ElementSetTool should steer people towards the right way to modify the agenda.